deployment/deploy/service.go

```diff
@@ -117,15 +118,15 @@ func (s *service) checkIfDeployedOnSystem(context *endly.Context, target *url.Re
 		return MatchVersion(request.Version, actualVersion), nil
 	}
 
-	transferTarget, err := context.ExpandResource(deploymentTarget.Deployment.Transfer.Dest)
+	dest, storageOpts, err := storage.GetResourceWithOptions(context, deploymentTarget.Deployment.Transfer.Dest)
 	if err != nil {
 		return false, err
 	}
-	service, err := storage.GetStorageService(context, transferTarget)
+	service, err := storage.StorageService(context, dest)
```

Comment:
you can use err from `dest, storageOpts, err := storage.GetResourceWithOptions(context, deploymentTarget.Deployment.Transfer.Dest)` instead of creating new one.

Reply:
generally, I could look closer into all err reusability, with explicit var OtherType declaration